lib/ecom/api.ts

```diff
@@ -290,6 +290,15 @@ function createApi(): EcomAPI {
                 return failureResponse(EcomApiErrorCodes.GetProductsFailure, getErrorMessage(e));
             }
         },
+        async getAllProducts(limit = 100) {
```

Comment:
this is not going to return all projects...
in fact, SDK/API could not return more then 100 items. if there is more then 100 products, there will be an issue. to handle this correctly, we have to load items in few requests, see example here:
https://github.com/codux-templates/e-commerce-remix/commit/fb0076f6093ed1cfeebbe0d96b3596ae107830d1

Reply:
@olehrakwix it does not need to return all products, it needs to return some products.

We could use `await api.getProductsByCategory('all-products');` instead. I'm not a huge fan of hardcoding the string `"all-products"` but it's hardcoded in many places.

Reply:
ok, I was under impression that we are going to collect all available on site routes with this function, but if we need just "some" products - then you are correct

Reply:
I'll consult with Omer and Nadav about this behavior and update here.

Reply:
I checked `api.getProductsByCategory`, it makes two requests: category first, and then the list of products in it. I kept `getAllProducts`, but renamed it to `getProducts`. I think I'll combine these two functions later, and make category an optional parameter.